Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1

```diff
@@ -153,7 +162,7 @@ function Get-SpnAccessToken {
     try
     {
         $proxyUri = Get-ProxyUri $authUri
-        if (($proxyUri -eq $null) -or ($proxyUri.AbsoluteUri -eq $null) -or ($proxyUri.AbsoluteUri -eq $authUri))
+        if (($proxyUri -eq $null) -or ($proxyUri.AbsoluteUri -eq $null) -or ($authUri -like "*$($proxyUri.Host)*"))
```

Comment:
I am checking that the target url has the same base host name as that of the proxy url. For ex, if the target url is http://xyz.com/some/path and the proxy url is http://xyz.com, then we do not need to use proxy.